File: src/main/java/org/cubectl/identity/role/RoleRepository.java
package org.cubectl.identity.role;

import java.util.Optional;
import java.util.UUID;

import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

public interface RoleRepository extends JpaRepository<RoleEntity, UUID> {

    Optional<RoleEntity> findByCode(String code);

    boolean existsByCode(String code);

    @EntityGraph(attributePaths = "permissions")
    Optional<RoleEntity> findWithPermissionsById(UUID id);
}

File: src/main/kotlin/org/cubectl/identity/role/RoleRepository.kt
package org.cubectl.identity.role

import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

interface RoleRepository : JpaRepository<RoleEntity, UUID> {

    fun existsByCode(code: String): Boolean

    @EntityGraph(attributePaths = ["permissions"])
    fun findByCode(code: String): Optional<RoleEntity>

    @EntityGraph(attributePaths = ["permissions"])
    fun findWithPermissionsById(id: UUID): Optional<RoleEntity>
}
